core/util.ts

```diff
@@ -12,14 +14,26 @@ export function isSemanticsTyped(json: any) {
 	return false;
 }
 
+export function requiresID(sType: string): boolean {
+	return sType.indexOf("definition") !== -1;
```

Comment:
Would you see another way of expressing this, preferably in a way that's not convention- (but rather metadata-)based?